aten/src/ATen/DLConvertor.cpp

```diff
@@ -57,19 +58,20 @@ static DLContext getDLContext(const Type& type, const int64_t& device_id) {
 }
 
 
-static Backend getATenBackend(const DLContext& ctx) {
-  Backend backend;
+static DeviceType getATenDeviceType(const DLContext& ctx) {
   switch (ctx.device_type) {
     case DLDeviceType::kDLCPU:
-      backend = Backend::CPU;
-      break;
+      return DeviceType::CPU;
     case DLDeviceType::kDLGPU:
-      backend = Backend::CUDA;
-      break;
+      return DeviceType::CUDA;
+    case DLDeviceType::kDLOpenCL:
+      return DeviceType::OPENCL;
+    case DLDeviceType::kDLROCM:
+      return DeviceType::HIP;
     default:
       throw std::logic_error("Unsupported device_type: " + std::to_string(ctx.device_type));
   }
-  return backend;
+  return DeviceType::CPU; // impossible
```

Comment:
Does impossible mean it should throw an error?

Reply:
I'm mostly traumatized from crappy C++ compilers not being able to tell that this case is in fact impossible and then tripping a Werror build, so I added this case to shut them up.

Reply:
Makes sense. I'd just add that as a comment.